src/logic.rs

```diff
@@ -88,25 +77,143 @@ pub fn get_move(game: &Game, _turn: &u32, board: &Board, me: &Battlesnake) -> &'
     return chosen;
 }
 
-fn valid_move(spot: &Coord, board: &Board, me: &Battlesnake) -> bool {
+fn spot_has_snake(spot: &Coord, snakes: &Vec<Battlesnake>) -> bool {
+    let mut snake_parts = vec![];
+    for snake in snakes {
+        snake_parts.push(snake.head);
+        snake_parts.append(&mut snake.body.clone());
```

Comment:
if you are passing a mut ref to a clone, why not just pass a copy?

Reply:
I feel like I've tried what I think this would look like, and it did not compile. Feel free to leave a code suggestion, and I'll try it.

Reply:
I was being a complete muppet because I'm tired - I was thinking if it was you're own written function you could change the signature but obviously its a vector...